File: src/Client/Logistics.OfficeApp/src/app/core/api/models/trip/update-trip.command.ts
import { CreateTripLoadCommand } from "./create-trip-load.command";
import { TripStopDto } from "./trip-stop.dto";

export interface UpdateTripCommand {
  tripId: string;
  name?: string;
  truckId?: string;
  newLoads?: CreateTripLoadCommand[] | null;
  attachedLoadIds?: string[] | null;
  detachedLoadIds?: string[] | null;
  optimizedStops?: TripStopDto[] | null;
}

File: src/Core/Logistics.Application/Commands/Trip/UpdateTrip/UpdateTripCommand.cs
using Logistics.Application.Abstractions;
using Logistics.Shared.Models;

namespace Logistics.Application.Commands;

public class UpdateTripCommand : IAppRequest
{
    public Guid TripId { get; set; }
    public string? Name { get; set; }
    public Guid? TruckId { get; set; }

    public IEnumerable<CreateTripLoadCommand>? NewLoads { get; set; }
    public IEnumerable<Guid>? AttachedLoadIds { get; set; }
    public IEnumerable<Guid>? DetachedLoadIds { get; set; }
    public IEnumerable<TripStopDto>? OptimizedStops { get; set; }
}
